### PR TITLE
Simplify the gene mention extractor

### PR DESCRIPTION
**Pull request by amwenger**
Improve the gene mention extractor by simplication.  Instead of
special cases, have one entity identification rule and two simple
supervision rules: true positives come from PMID<->gene mappings
from NCBI, and true negatives come from uppercase words that
are not gene symbols.

### CODE REVIEW

File: code/gene_extract_candidates.py [hunk 1]
```diff
@@ -8,46 +9,40 @@
...
-    gene_bigrams = set([x.strip().lower() for x in f])
-  bad_genes = gene_english | gene_bigrams
-  return bad_genes
-
-
-def read_misc_noisy_genes():
-  """Read Feng's manually compiled list of weird gene names
+      phrase,ensembl_id,mapping_type = line.rstrip('\n').split('\t')
+      phrase_to_genes[phrase].add((ensembl_id,mapping_type))
+      lower_phrase_to_genes[phrase].add((ensembl_id,mapping_type))
```
Comment (robinjia) [inline code/gene_extract_candidates.py]: Do you mean phrase.lower() ?

File: code/gene_extract_candidates.py [hunk 2]
```diff
@@ -90,25 +82,18 @@ def get_supervision(row, mention):
...
-  # Here, require at least 3 letters and 1 digit.
-  if mention.mention_type in ('NAME', 'SYN'):
-    # An exact case match
-    if re.match(r'[a-zA-Z]{3}[a-zA-Z]*\d+\w*', word):
+  # Negative Rule #1: words that are all uppercase but are not in the gene symbol
+  # list are likely false mentions.
+  if mention.mention_type in ('ALL_UPPERCASE_NOT_GENE_SYMBOL', 'RANDOM_WORD_NOT_GENE_SYMBOL'):
+    return False
+
+  # Positive Rule #2: matches from papers that NCBI annotates as being about
```
Comment (robinjia) [inline code/gene_extract_candidates.py]: Positive Rule #1?

File: code/gene_extract_candidates.py (hunk 1 above)
Comment (amwenger) [inline code/gene_extract_candidates.py]: Good catch.  I'll fix that.

File: code/gene_extract_candidates.py (hunk 2 above)
Comment (amwenger) [inline code/gene_extract_candidates.py]: Cleaning it up...

**Pull request merged**